### PR TITLE
minor textual changes to README and dedicated example

### PR DESCRIPTION
**Pull request by fantapop**
Including:
* sync variables mentioned in README with actual example
* add example values for allow list name and CIDR IP
* remove limited access denotion for Azure clusters
* add link to terraform sensitive values recommendations in example comment.

**Commit checklist**
- [x] Changelog
- [x] Doc gen (`make generate`)
- [ ] Integration test(s)
- [ ] Acceptance test(s)
- [ ] Example(s)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Realign variables used in the cockroach_dedicated_cluster with the README
+
+- Removed mention of Limited Access for Azure clusters in README
+
+- Added some example values for clarity in README
+
```
Comment (marksoper) [inline CHANGELOG.md:17]: I assume you're leaving out the change of sensitive values intentionally?

File: README.md [hunk 2]
```diff
@@ -130,13 +130,13 @@ Before you use `terraform-provider-cockroach` you must [install Terraform](https
...
+        - `<number of VCPUs per node>` is the number of virtual CPUS assigned to each node.  This number will dictate which machine type is used for your cluster nodes. Recommendations for choosing this value can be found [here](https://www.cockroachlabs.com/docs/cockroachcloud/create-your-cluster#step-5-configure-cluster-capacity).  
         - `<storage in GiB>` is the amount of storage specified in GiB.  
-        - `<cloud provider machine type>` is the machine type for the cloud infrastructure provider.  
-        - `<allow list name>` is the name for the IP allow list. Use a descriptive name to identify the IP allow list.  
-        - `<allow list CIDR IP>` is the Classless Inter-Domain Routing (CIDR) IP address base.  
+        - `<allow list name>` is the name for the IP allow list. Use a descriptive name to identify the IP allow list. (i.e. "allow all" or "home network")  
+        - `<allow list CIDR IP>` is the Classless Inter-Domain Routing (CIDR) IP address base. (i.e. 123.123.123.123)
```
Comment (marksoper) [inline README.md:139]: NIT: would adding a mask be more representative of CIDR notation? E.g. 198.51.100.14/24

**Review by marksoper — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (fantapop) [reply README.md:139]: Our allowlist resource asks for the ip portion and the prefix length separately.  Unfortunately we also use the name cidr_mask to refer to the prefix length.   I've tried to clear that up with some documentation changes.  I wouldn't be opposed to deprecating the existing eventually and using a single string field for the full cidr notation as a single string.

File: CHANGELOG.md (hunk 1 above)
Comment (fantapop) [reply CHANGELOG.md:17]: yeah, it just doesn't seem like something that needs to go in the changelog.

**Pull request merged**